File: order-service/src/main/java/com/ecommerce/orderservice/client/product/ProductClient.java
package com.ecommerce.orderservice.client.product;

import com.ecommerce.orderservice.client.product.dto.ProductNameResponse;
import com.ecommerce.orderservice.client.product.dto.ProductPriceResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@FeignClient(name = "product-service", fallbackFactory = ProductClientFallbackFactory.class)
public interface ProductClient {

    @GetMapping("/api/v1/product/price")
    ProductPriceResponse getPriceMap(@RequestParam List<Long> productIds);

    @GetMapping("/api/v1/product/names")
    ProductNameResponse getNamesMap(@RequestParam List<Long> productIds);
}

File: review-service/src/main/kotlin/com/ecommerce/reviewservice/client/product/ProductClient.kt
package com.ecommerce.reviewservice.client.product

import com.ecommerce.reviewservice.client.product.dto.ProductNameResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient("product-service", fallbackFactory = ProductClientFallbackFactory::class)
interface ProductClient {

    @GetMapping("/api/v1/product/names")
    fun getProductNames(@RequestParam productIds: List<Long>): ProductNameResponse
}
